### PR TITLE
feat: add kubo-rpc-client

### PR DESCRIPTION
**Pull request by SgtPooki**
fixes https://github.com/ipfs/js-kubo-rpc-client/issues/4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -181,7 +181,7 @@
     "import-meta-resolve": "^2.0.3",
     "ipfs-unixfs": "^7.0.0",
     "ipfs-utils": "^9.0.1",
-    "ipfsd-ctl": "^12.0.0",
+    "ipfsd-ctl": "github:sgtpooki/js-ipfsd-ctl#feature/add-js-kubo-rpc-client",
```
Comment (SgtPooki) [inline package.json:184]: will replace with proper version when https://github.com/ipfs/js-ipfsd-ctl/pull/764 is merged and published.

**Comment by SgtPooki** [comment]
Currently getting an error on `js-rv1-js` circuit test. see attached log from running `DEBUG='ipfs*,libp2p*' GOLOG_LOG_LEVEL='debug,core/server=debug' npm run test -- -g 'js-rv1-js' -d &> js-rv1-js.log`
[js-rv1-js.log](https://github.com/ipfs/interop/files/9601471/js-rv1-js.log)

File: test/cid-version-agnostic.js [hunk 2]
```diff
@@ -32,11 +32,17 @@ describe('CID version agnostic', function () {
     Object.assign(daemons, { js0, js1, go0, go1 })
 
     await Promise.all([
+      // @ts-ignore
```
Comment (achingbrain) [inline test/cid-version-agnostic.js:35]: It's generally better to use `@ts-expect-error` as we usually add an ignore as a temporary measure but they have a way of sticking around forever.  `expect-error` will start to fail if the underlying issue is fixed.

That said, what is the underlying issue that needed all these ignores adding?

Comment (achingbrain) [reply test/cid-version-agnostic.js:35]: Is it because this PR needs merging? https://github.com/ipfs/js-kubo-rpc-client/pull/25

Comment (SgtPooki) [reply test/cid-version-agnostic.js:35]: yea let me look at that PR

**Review by SgtPooki — COMMENTED** [review]
self review

**Review by SgtPooki — COMMENTED** [review]
self review

File: package.json [hunk 3]
```diff
@@ -181,7 +181,7 @@
     "import-meta-resolve": "^2.0.3",
     "ipfs-unixfs": "^7.0.0",
     "ipfs-utils": "^9.0.1",
-    "ipfsd-ctl": "^12.0.0",
+    "ipfsd-ctl": "https://gitpkg.now.sh/SgtPooki/js-ipfsd-ctl?feature/add-js-kubo-rpc-client&scripts.postinstall=npm%20install%20--ignore-scripts%20%26%26%20npm%20run%20build",
```
Comment (SgtPooki) [inline package.json:184]: will replace with proper version when https://github.com/ipfs/js-ipfsd-ctl/pull/764 is merged and published.

**Comment by SgtPooki** [comment]
Updated package-lock.json so npm install stops failing: https://github.com/ipfs/interop/actions/runs/3114818422/jobs/5112612445

**Comment by BigLep** [comment]
@SgtPooki : we can remove the "status/blocked" label right?

@achingbrain : given CI is green, is it ok to merge?

**Comment by achingbrain** [comment]
https://github.com/ipfs/js-ipfsd-ctl/pull/764 needs to land first, then all good here.

**Comment by SgtPooki** [comment]
Should be able to update this with the newest version once `js-ipfsd-ctl` releases: https://github.com/ipfs/js-ipfsd-ctl/actions/runs/3192111210

**Comment by SgtPooki** [comment]
version 12.12.0 is pushed to npm: https://www.npmjs.com/package/ipfsd-ctl

**Pull request merged**